### DISCUSSION TITLE
Graphing in Textual?

### DISCUSSION THREAD

**Discussion by misterek** [Textualize/textual#3260] [General]
Is there any pattern so far for graphing things in textual?  I've been playing around with another library to create graphs, but I suspect that there are terminal codes that aren't completely compatible. For instance:

```
class TesterApp(App):

    CSS_PATH = "tester.css"

    BINDINGS = [("q", "quit", "Quit")]

    def compose(self) -> ComposeResult:

        yield Header()

        y = plt.sin() # sinusoidal test signal
        plt.scatter(y)
        plt.title("Scatter Plot") # to apply a title
        plt.plot_size(100,100)

        s = Static(Text(plt.build()))
        yield s

        yield Footer()

    def action_quit(self) -> None:
        self.app.exit("quit")

if __name__ == "__main__":
    app = TesterApp()
    app.run()
```

```
Screen > Static {
    background: green;
    color: white;
}
```

Using https://github.com/piccolomo/plotext.  

This leads to a... strange looking graph.  Interestingly, if I add width: 500 to the css, it displays the graph correctly, with a few straggling lines off to the side.  (Wrapping it in a Rich Text was an attempt to get it to work, but has the same result)

Has anyone had any luck with this or a similar library for graphing?

With no width:
![image](https://github.com/Textualize/textual/assets/18128619/884e3ac3-5929-4d39-b20e-09ded1961b9c)

With width: 500
![image](https://github.com/Textualize/textual/assets/18128619/d5a4e859-bfd7-47b4-a948-94ee4d117e50)

I've played around with the different options as well as I think I can, but can't quite get it to display like I hope it would.  

Maybe there are other libraries that might work better?

**Comment by TomJGooding** [comment]
It looks like adding a Textual wrapper for Plottext is planned - see #3029

**Reply by misterek** [reply]
🤦   Apparently my searching abilities are terrible.  Thank you for the pointer.  Closing this as 'I'm an idiot'.

**Reply by davep** [reply]
Meanwhile, you may find this conversation on Discord interesting too: https://discord.com/channels/1026214085173461072/1033752599112994867/1138912949474439309
